### PR TITLE
Feat/dataparsing

### PR DESCRIPTION
**Pull request by Sinduja98**
### Functionality Implemented:

- **LLM Response Parsing**: Parses each chunk of the LLM response.
- **Tool Use**:
  - `write_to_file`: Creates a new file and places the parsed content into it.
  - `execute_command`: Opens a new terminal and executes the parsed LLM response as a command.

### Output:

- The parsed LLM response is written to a file.
- When prompted by the user, code or commands are executed in the terminal.

### Known Issues and TODOs:

- Added comments to address specific edge cases in upcoming updates.
- The backend should keep the user updated with status messages. A success message wrapped inside `<attempt_completion>` should be shown upon completion.

### CODE REVIEW

**Review by adithya1012 — APPROVED** [review]
Working in my local

File: src/core/index.ts [hunk 1]
```diff
@@ -135,6 +170,173 @@ export class DevAssist {
...
+							.split("")
+							.map((char) => `(?:${char})?`)
+							.join("")}$`,
+						"g"
+					)
+					return text.replace(tagRegex, "")
+				}
+				switch (block.name) {
+					case "read_file": {
+						//TODO: implement read_file
```
Comment (hdprajwal) [inline src/core/index.ts:267]: Can you add appropriate break statements for switch cases that are yet to be implemented.
This is causing issues when running the code with tool use

File: src/core/index.ts [hunk 2]
```diff
@@ -135,6 +170,173 @@ export class DevAssist {
...
+					case "execute_command": {
+						const command: string = block.params.command
+								if (!command) {
+									pushToolResult(
+										await this.sayAndCreateMissingParamError("execute_command", "command")
+									)
+									break
+								}
+								const [userRejected, result] = await this.executeCommandTool(command)
+								pushToolResult(result)
```
Comment (hdprajwal) [inline src/core/index.ts:279]: Add break statement for this swtich case after the result of the Tool is pushed.

File: src/core/index.ts [hunk 3]
```diff
@@ -135,6 +170,173 @@ export class DevAssist {
...
+								content = content.slice(0, lastOpenBracketIndex).trim()
+							}
+						}
+					}
+				}
+				await this.say("text", content)
+				break
+			}
+			case "tool_use":
+				const toolDescription = () => {
```
Comment (hdprajwal) [inline src/core/index.ts:209]: Can you handle the default case  for switch case when none of the cases match. 
If the LLM happens to produce a block that does not match any of the listed cases due to hallucination then we would want to handle that case and not throw an error.

File: src/core/index.ts [hunk 4]
```diff
@@ -135,6 +170,173 @@ export class DevAssist {
...
+					}
+				}
+		}
+		if (!block.partial ) {
+			this.currentStreamingContentIndex++ 
+		} 
+	}
+
+	async executeCommandTool(command: string): Promise<[boolean,string]> {
+		const terminalInfo = await this.terminalManager.getOrCreateTerminal(cwd)
```
Comment (nidhitalastha) [inline src/core/index.ts:325]: getOrCreateTerminal is not reusing the existing terminal and always creates a new terminal for execution.

**Review by jahnavib26 — APPROVED** [review]

File: src/core/index.ts (hunk 4 above)
Comment (Sinduja98) [reply src/core/index.ts:325]: In the future work, we can have Terminal Registry and have a track of the terminal.

File: src/core/index.ts (hunk 1 above)
Comment (Sinduja98) [reply src/core/index.ts:267]: Updated the code by adding the break statements for each case.

File: src/core/index.ts (hunk 2 above)
Comment (Sinduja98) [reply src/core/index.ts:279]: Yes, it is added.

File: src/core/index.ts (hunk 3 above)
Comment (Sinduja98) [reply src/core/index.ts:209]: Updated the code to handle 'default' case.

**Review by hdprajwal — APPROVED** [review]

**Review by nidhitalastha — APPROVED** [review]

**Pull request merged**